src/lib/classes/Imap.php

```diff
@@ -294,4 +300,47 @@ public function isOnlyFolderMessageNum($folderNum, $folderMessageNum) {
 		}
 		return in_array($folderMessageNum, $onlyFolderMessagesNum);
 	}
+
+	public function isMessageFlagsEnabled() {
+		$conf = $this->getConf();
+
+		if (isset($conf['flags']) && !empty($conf['flags'])) {
+			return true;
+		}
+		else
+			return false;
+	}
+
+	public function getFolderMessagesFlags($length) {
+		$messagesFlags = array();
+		$FLAGS_OFFSET = 44;
+
+		$headers = imap_headers($this->getConnection());
+		if ($headers == true) {
+			foreach ($headers as $header) {
+				$flags = NULL;
+
+				if (preg_match("/{.*?}/", $header, $matches, PREG_OFFSET_CAPTURE, $FLAGS_OFFSET) == true) {
+					foreach ($matches as $match) {
+						if ($match[1] != $FLAGS_OFFSET) {
+							continue;
+						}
+
+						$flags = ltrim($match[0], "{");
+						$flags = rtrim($flags, "}");
+						break;
+					}
+				}
+
+				$messagesFlags[] = $flags;
+			}
+		}
+
+		// Need array to start at 1 to match message indices
+		array_unshift($messagesFlags, "dummy");
```

Comment:
i'd prefer to use NULL instead of "dummy" :)

Reply:
Hah, yes, good call...